### PR TITLE
Feature/java example

### PR DESCRIPTION
**Pull request by luckychess**
## How to use the features provided in the pull request?
Go to example folder.
```
./prepare.sh
gradle build
gradle run
```

### CODE REVIEW

File: example/java/TransactionExample.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    public static void main(String[] args) {
+        String adminPub = "", adminPriv = "";
+
+        try {
+            adminPub = new String(Files.readAllBytes(Paths.get("../admin@test.pub")));
+            adminPriv = new String(Files.readAllBytes(Paths.get("../admin@test.priv")));
+        } catch (IOException e) {
+            System.err.println("Unable to read key files.\n " + e);
+            System.exit(1);
+        }
```
Comment (Warchant) [inline example/java/TransactionExample.java]: can this code be added to `irohajava` package?
so the user writes something like `irohajava.load()`

File: example/java/prepare.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+cd $(dirname $0)
+
+# generate iroha lib
+if [[ $1 -eq 0 && ${IROHA_HOME} -eq 0 ]] ; then
```
Comment (Warchant) [inline example/java/prepare.sh]: "$1", "${IROHA_HOME}"

in what case IROHA_HOME will be equal to 0? 

for emptiness better use -z

```
if [ -z "$1" && -z "${IROHA_HOME}" ]; then
```

File: example/java/prepare.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    echo 'Neither IROHA_HOME nor path to CMakeLists.txt is set'
+    exit 0
+fi
+cmake_path=${1,-${IROHA_HOME}}
```
Comment (Warchant) [inline example/java/prepare.sh]: what a weird construction?
At what cases it is required to set `cmake_path` to 0?

File: example/java/TransactionExample.java (hunk 1 above)
Comment (luckychess) [reply example/java/TransactionExample.java]: User loads library via ```System.loadLibrary("irohajava");``` but how can library be aware of user's keys and their location?

File: example/java/build.gradle [hunk 4]
```diff
@@ -0,0 +1,61 @@
+apply plugin: 'java'
+apply plugin: 'com.google.protobuf'
+apply plugin: 'application'
+
+def BUILD_DIR = "./shared_model/bindings"
```
Comment (kamilsa) [inline example/java/build.gradle]: not build/shared_model/bindings ?

File: example/java/build.gradle [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.3'
+    }
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    compile 'com.google.protobuf:protobuf-java:3.4.0'
```
Comment (l4l) [inline example/java/build.gradle]: should be 3.5.0

File: example/java/build.gradle [hunk 6]
```diff
@@ -0,0 +1,61 @@
+apply plugin: 'java'
+apply plugin: 'com.google.protobuf'
```
Comment (l4l) [inline example/java/build.gradle]: should be after buildscript

File: example/java/TransactionExample.java [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+import java.nio.file.Paths;
+import java.nio.file.Files;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.lang.Thread;
+
+class TransactionExample {
+    static {
+        try {
+            System.loadLibrary("irohajava");
```
Comment (l4l) [inline example/java/TransactionExample.java]: Imo we should make a package that doesn't require manual loading

File: example/java/TransactionExample.java [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+
+    private static ModelCrypto crypto = new ModelCrypto();
+    private static ModelTransactionBuilder txBuilder = new ModelTransactionBuilder();
+    private static ModelQueryBuilder queryBuilder = new ModelQueryBuilder();
+    private static ModelProtoTransaction protoTxHelper = new ModelProtoTransaction();
+    private static ModelProtoQuery protoQueryHelper = new ModelProtoQuery();
+
+
+
+    public static void main(String[] args) {
```
Comment (l4l) [inline example/java/TransactionExample.java]: Would be nice to extract functions from main. E.g `readKeyFrom`, `bytesFromVector`, etc

File: example/java/TransactionExample.java (hunk 7 above)
Comment (luckychess) [reply example/java/TransactionExample.java]: Hmm, I have no idea how to to this. In Python we use exactly the same approach.

File: example/java/build.gradle (hunk 6 above)
Comment (luckychess) [reply example/java/build.gradle]: Why not put all all "apply" in one place?

Comment (l4l) [reply example/java/build.gradle]: Hm, I thought it doesn't work, sorry

**Pull request merged**